hyperspy/_signals/lazy.py

```diff
@@ -521,87 +551,98 @@ def _map_all(self, function, inplace=True, **kwargs):
+                                                                        ragged=ragged,
+                                                                        **testing_kwargs)
+        # Dropping/Adding Axes
+        if output_signal_size == self.axes_manager.signal_shape:
+            drop_axis = None
+            new_axis = None
+            axes_changed = False
+        else:
+            axes_changed = True
+            if len(output_signal_size) != len(self.axes_manager.signal_shape):
+                drop_axis = self.axes_manager.signal_indices_in_array
+                new_axis = tuple(range(len(output_signal_size)))
             else:
-                cm = dummy_context_manager
-            with cm():
-                try:
-                    for i, pixel in enumerate(pixels):
-                        res_data.flat[i] = pixel.compute()
-                except MemoryError:
-                    raise MemoryError("The use of 'ragged' array requires the "
-                                      "computation of the lazy signal.")
+                drop_axis = [it for (o, i, it) in zip(output_signal_size,
+                                                      self.axes_manager.signal_shape,
+                                                      self.axes_manager.signal_indices_in_array)
+                             if o != i]
+                new_axis = drop_axis
+        chunks = tuple([self.data.chunks[i] for i in sorted(nav_indexes)]) + output_signal_size
+        mapped = da.map_blocks(process_function_blockwise,
+                               self.data,
+                               *args,
+                               function=function,
+                               nav_indexes=nav_indexes,
+                               drop_axis=drop_axis,
+                               new_axis=new_axis,
+                               output_signal_size=output_signal_size,
+                               dtype=output_dtype,
+                               chunks=chunks,
+                               **kwargs)
+        if inplace:
+            self.data = mapped
+            sig = self
         else:
-            if len(pixels) > 0:
-                for step in reversed(res_shape):
-                    _len = len(pixels)
-                    starts = range(0, _len, step)
-                    ends = range(step, _len + step, step)
-                    pixels = [
-                        da.stack(
-                            pixels[s:e], axis=0) for s, e in zip(starts, ends)
-                    ]
-            res_data = pixels[0]
-
-        res = map_result_construction(
-            self, inplace, res_data, ragged, sig_shape, lazy=not ragged)
-
-        return res
+            sig = self._deepcopy_with_new_data(mapped)
+            if ragged:
+                sig.axes_manager.remove(sig.axes_manager.signal_axes)
+                sig.__class__ = LazySignal
+                sig.__init__(**sig._to_dictionary(add_models=True))
```

Comment:
To assign to the correct class, replace with:
```suggestion
                sig._lazy = True
                sig._assign_subclass()
```